### PR TITLE
ocrvs-3552: Use operation history to show last activity time in workqueue

### PR DESCRIPTION
**Pull request by rabiulislamanik**
The description of #3552  said, FA "Sent for review" column in "Sent for review" tab doesn't show original time of submission.

Similar type of issues were also occurring in other workqueue tabs too. The reasons behind these were - 
* We have been using  the "modifiedAt' (or 'createdAt' as fallback) field to know last activity(submission,rejection,approval etc). So whenever a declaration is modified, that time of latest modification is reflected in every others tabs for that declaration. Thus, when a declaration is rejected, the Field Agent sees 'the time of rejection' of that declration in his 'sent for review' Tab.
* The createdAt field was being changed everytime while upserting the event  in elasticSearch.

What has been changed in this PR - 
* Prevented createdAt from being changed everytime event has been updated.
* Used operationHistories to get latest activity time by activity type.
* Saving operationHistories for declared state from now on.
* There was another existing bug, which used to cause 'operationHistory.operatedOn' time to be inaccurate(the previous modifiedAt time was not changing) when a declaration gets rejected. It has been fixed now.

### CODE REVIEW

**Comment by euanmillar** [comment]
We have decided that the datetimes associated with these events while incorrect in previous versions do not represent a large enough user experience problem to warrant a migration script to be written

File: packages/gateway/src/features/search/type-resolvers.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@
  * graphic logo are (registered/a) trademark(s) of Plan International.
  */
 import { NATIVE_LANGUAGE } from '@gateway/constants'
-import { GQLDate, GQLResolver } from '@gateway/graphql/schema'
+import {
+  // GQLDate,
```
Comment (Zangetsu101) [inline packages/gateway/src/features/search/type-resolvers.ts:14]:
```suggestion
```

File: packages/search/src/elasticsearch/utils.ts [hunk 2]
```diff
@@ -403,7 +403,7 @@ export function isValidOperationHistory(body: IBirthCompositionBody) {
   const validStatusMapping = {
     [ARCHIVED_STATUS]: [DECLARED_STATUS, REJECTED_STATUS, VALIDATED_STATUS],
     [IN_PROGRESS_STATUS]: [null],
-    [DECLARED_STATUS]: [ARCHIVED_STATUS],
+    [DECLARED_STATUS]: [ARCHIVED_STATUS, IN_PROGRESS_STATUS, null],
```
Comment (Zangetsu101) [inline packages/search/src/elasticsearch/utils.ts:406]:
```suggestion
    [DECLARED_STATUS]: [ARCHIVED_STATUS, null],
```
I don't think we can go from IN_PROGRESS_STATUS to DECLARED_STATUS or can we?

File: packages/client/src/views/OfficeHome/sentForReview/SentForReview.tsx [hunk 3]
```diff
@@ -216,9 +218,11 @@ class SentForReviewComponent extends React.Component<
...
         ''
       const sentForApproval =
-        (reg.createdAt && Number.isNaN(Number(reg.createdAt))
-          ? new Date(reg.createdAt)
-          : new Date(Number(reg.createdAt))) || ''
+        getPreviousOperationDateByOperationType(
+          reg.operationHistories,
+          this.isFieldAgent ? RegStatus.Declared : RegStatus.Validated
+        ) || ''
+
```
Comment (Zangetsu101) [inline packages/client/src/views/OfficeHome/sentForReview/SentForReview.tsx:225]: @euanmillar For declarations that don't have `operationHistories`, should we show some fallback value(which would be wrong) or not show anything at all?

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**